platform/pddf/i2c/utils/pddf_util.py

```diff
@@ -0,0 +1,450 @@
+    switch-nonpddf  : switch to per platform, non-pddf mode
+"""
+
+import os
+import commands
+import sys, getopt
+import logging
+import re
+import subprocess
+import shutil
+import time
+import json
+import pddfparse
+from collections import namedtuple
+
+PLATFORM_ROOT_PATH = '/usr/share/sonic/device'
+SONIC_CFGGEN_PATH = '/usr/local/bin/sonic-cfggen'
+HWSKU_KEY = 'DEVICE_METADATA.localhost.hwsku'
+PLATFORM_KEY = 'DEVICE_METADATA.localhost.platform'
+
+PROJECT_NAME = 'PDDF'
+version = '1.1'
+verbose = False
+DEBUG = False
+args = []
+ALL_DEVICE = {}               
+FORCE = 0
+
+# Instantiate the class pddf_obj 
+pddf_obj = pddfparse.PddfParse()
+try:
+    import pddf_switch_svc
+except ImportError:
+    print "Unable to find pddf_switch_svc.py. PDDF might not be supported on this platform"
+    sys.exit()
+
+
+
+if DEBUG == True:
+    print sys.argv[0]
+    print 'ARGV      :', sys.argv[1:]   
+
+def main():
+    global DEBUG
+    global args
+    global FORCE
+        
+    if len(sys.argv)<2:
+        show_help()
+         
+    options, args = getopt.getopt(sys.argv[1:], 'hdf', ['help',
+                                                       'debug',
+                                                       'force',
+                                                          ])
+    if DEBUG == True:                                                           
+        print options
+        print args
+        print len(sys.argv)
+            
+    for opt, arg in options:
+        if opt in ('-h', '--help'):
+            show_help()
+        elif opt in ('-d', '--debug'):            
+            DEBUG = True
+            logging.basicConfig(level=logging.INFO)
+        elif opt in ('-f', '--force'): 
+            FORCE = 1
+        else:
+            logging.info('no option')                          
+    for arg in args:            
+        if arg == 'install':
+            do_install()
+        elif arg == 'clean':
+           do_uninstall()
+        elif arg == 'switch-pddf':
+            do_switch_pddf()
+        elif arg == 'switch-nonpddf':
+            do_switch_nonpddf()
+        else:
+            show_help()
+            
+    return 0              
+        
+def show_help():
+    print __doc__ % {'scriptName' : sys.argv[0].split("/")[-1]}
+    sys.exit(0)
+
+def my_log(txt):
+    if DEBUG == True:
+        print "[PDDF]"+txt    
+    return
+    
+def log_os_system(cmd, show):
+    logging.info('Run :'+cmd)  
+    status, output = commands.getstatusoutput(cmd)    
+    my_log (cmd +"with result:" + str(status))
+    my_log ("      output:"+output)    
+    if status:
+        logging.info('Failed :'+cmd)
+        if show:
+            print('Failed :'+cmd)
+    return  status, output
+            
+def driver_check():
+    ret, lsmod = log_os_system("lsmod| grep pddf", 0)
+    logging.info('mods:'+lsmod)
+    if len(lsmod) ==0:
+        return False   
+    return True
+
+kos = [
+'modprobe i2c-ismt',
```

Comment:
This lis of ko's include linux specific dirvers and also pddf specific drivers. can we not keep it separate ?
Also allow users to add in /etc/module.d/generic.conf  and /etc/module.d/pddf.conf  ?

Reply:
Separating the pddf drivers and linux drivers (which might differ based on platform) is already in plan for PDDF 2.0, where we will be moving to 2.0 Platform APIs implementation as well.